cmake/libigl.cmake

```diff
@@ -414,13 +414,16 @@ endif()
 if(LIBIGL_WITH_PNG)
   # png/ module is anomalous because it also depends on opengl it really should
   # be moved into the opengl/ directory and namespace ...
+  if(NOT TARGET stb_image)
+    igl_download_stb()
+    add_subdirectory(${LIBIGL_EXTERNAL}/stb stb_image)
+  endif()
+  compile_igl_module("png" "")
+  target_link_libraries(igl_png ${IGL_SCOPE} igl_stb_image )
+  
   if(TARGET igl_opengl)
-    if(NOT TARGET stb_image)
-      igl_download_stb()
-      add_subdirectory(${LIBIGL_EXTERNAL}/stb stb_image)
-    endif()
-    compile_igl_module("png" "")
-    target_link_libraries(igl_png ${IGL_SCOPE} igl_stb_image igl_opengl)
+    compile_igl_module("png_opengl" "")
```

Comment:
Maybe we should have a CMake option `LIBIGL_WITH_PNG_OPENGL` for consistency with other modules.

Reply:
right now if you enable WITH_PNG and WITH_OPENGL you also get png_opengl , so no need to have an extra switch?